File: week-1/longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs or len(strs) == 0:
            return ''
        prefix = ''
        index = 0
        for i in strs[0]:
            for j in range(1, len(strs)):
                if index >= len(strs[j]) or i != strs[j][index]:
                    return prefix
            prefix += i
            index += 1
        
        return prefix

File: week-1/longest-common-prefix/longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    if (!strs || strs.length == 0) return '';
    let prefix = '';
    let index = 0;
    for (let i=0; i<strs[0].length; i++) {
        for (let j=1; j<strs.length; j++) {
            if (index >= strs[j].length || strs[0][i] != strs[j][index]) return prefix;
        }
        prefix += strs[0][i];
        index += 1;
    }
    return prefix
};
